### DISCUSSION TITLE
AEC Virtual Entities

### DISCUSSION THREAD

**Discussion by melihpeker-pointr** [mozman/ezdxf#591] [Q&A] [answered]
Hi! I'm trying to get the virtual entities in an `AEC_SPACE` entity. The original DXF file has an AEC_SPACE entity and I'm trying to get its virtual entities by using the recommendations that @mozman give in #495. 

Here's the problem. When I exploded the AEC_SPACE entity on AutoCAD the hierarchy is as follows:

`AEC_SPACE -> BLOCK REFERENCE -> HATCH -> POLYLINE`

However, when I used the following method for getting virtual entities I get directly the `POLYLINE`:
```python
proxy_entities = list(model_space.query("ACAD_PROXY_ENTITY"))
    for proxy_entity in proxy_entities:
        if proxy_entity.proxy_graphic:
            for entity in ProxyGraphic(proxy_entity.proxy_graphic, doc).virtual_entities():
                print('Virtual inside proxy is : ' + str(entity.dxftype()))
                model_space.add_entity(entity)
            proxy_entity.destroy()  # remove original proxy entity if not required anymore
```
Is there any way that I can get `BLOCK` or `HATCH` entities using `ProxyGraphic(proxy_entity.proxy_graphic, doc).virtual_entities()` function? 

I need `HATCH` entities inside the `ACAD_PROXY_ENTITY` since I don't want to draw them using drawing add-on. Since I get directly the POLYLINE, the drawing Frontend does not understand that they are hatch entities so `hatch_policy=HatchPolicy.IGNORE` is not working.

Also not using the `virtual_entities` is not an option since I need that function for other proxy entities. 

The corresponding DXF file is attached.
[aec_space_test.zip](https://github.com/mozman/ezdxf/files/7744118/aec_space_test.zip)

Thanks in advance.

**Comment by mozman** [comment, marked as answer]
The proxy-graphic is a simplified graphical representation of a DXF entity, so applications which do not know the entity can at least show something to the user. There is no documentation by Autodesk for the proxy-graphic. The OpenDesign Specification for DWG has a short documentation of the data structures used in proxy-graphic data and there are only line, arc, polyline, mesh, shell and text objects.

There are no real DXF entities stored in the proxy-graphic data, what you get as virtual entities by `ezdxf` is already the luxury version, there are no block references or hatches stored in the proxy-graphic data and the `drawing` add-on shows exact that content.

**Comment by melihpeker-pointr** [comment]
Ok, thank you for the clarification.
